### ISSUE TITLE
Re-enable `simdjson` test in `ci.yml`

### ISSUE THREAD

**Issue by scottbot95** [serenity-rs/poise#184]
The `simdjson` test was disabled in 1b0f83bcb0ddea24bf82ddfa3b84f3918be50ebb due to upstream issues with the `serenity/simdjson` (serenity-rs/serenity#2474). Once upstream issues are resolved, we should re-enable the `simdjson` test.

**Comment by jamesbt365** [comment]
We have removed simd-json in serenity-rs/serenity#2735 on the `next` branch. Is this still valid (current still has it but is there any purpose to getting it to work if it still doesn't)?

**Comment by scottbot95** [comment]
I don't think we need to re-enable this test anymore if we've removed simd-json entirely. Probably should cleanup the workflow though and any other references to simd-json  https://github.com/serenity-rs/poise/blob/6c26be9d3fa6ee092b9e418618596dfdf443097b/.github/workflows/ci.yml#L47
